src/Temporalio/Client/Configuration/ClientConfig.cs

```diff
@@ -0,0 +1,124 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.Json;
+
+namespace Temporalio.Client.Configuration
+{
+    /// <summary>
+    /// Represents the overall client configuration.
+    /// </summary>
+    public sealed class ClientConfig : ICloneable
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ClientConfig"/> class.
+        /// </summary>
+        /// <param name="profiles">The configuration profiles.</param>
+        public ClientConfig(IReadOnlyDictionary<string, ClientConfigProfile> profiles)
+        {
+            Profiles = profiles ?? new Dictionary<string, ClientConfigProfile>();
+        }
+
+        /// <summary>
+        /// Gets the configuration profiles.
+        /// </summary>
+        public IReadOnlyDictionary<string, ClientConfigProfile> Profiles { get; private set; }
```

Comment:
```suggestion
    public sealed record ClientConfig(IReadOnlyDictionary<string, ClientConfigProfile> profiles)
    {
```
I think this is better as a record instead of a simple class like our "options" classes are. And can remove cloneable since records have their own way to kinda clone. Same for the other options.